lib/cfg-lex.l

```diff
@@ -127,6 +127,23 @@ yy_input_run_backtick_substitution(CfgLexer *self, gchar *buf, gsize buf_size)
     longjmp(__yyg->yyextra_r->fatal_error, 1);				\
   } while(0)
 
+#define _read_until_eof_or_eol(scanner,content,cond)  \
+  do {                                                \
```

Comment:
I think this could become a function instead of a macro. The others are overriding lexer functionality, but this should just be a function. It does not even change the control flow.

Reply:
I would also change the verb "read" to something like, I don't have a firm opinion, the best I could come up with is  _process_commented_line(), or instead of process, it could be scan/parse/skip, however reading as a term does not apply.

This is probably nitpicking, but if you change this to a function anyway, it'd be easy to rename it as well.

Reply:
I had trouble with `input` / `unput`, when it was a function; otherwise I would it to be a function.
Sure renaming make sense.